Fix behavior of psql -c option with postgresql 9.6

maximeguillet opened the pull request:
### What does this PR do?

Fix problem with psql command call when a custom .psqlrc file is set for system user postgres.

In postgresql 9.6 release notes (https://www.postgresql.org/docs/9.6/static/release-9-6.html), it appear a modification has been done to the `-c` option:

> - psql's -c option no longer implies --no-psqlrc (Pavel Stehule, Catalin Iacob)
> 
> Write --no-psqlrc (or its abbreviation -X) explicitly to obtain the old behavior. Scripts so modified will still work with old versions of psql.

`--no-psqlrc` option must be explicitly set when psql is called.
### What issues does this PR fix or reference?
#36787
### Tests written?

No

maximeguillet committed:
Fix behavior of psql -c option with postgresql 9.6

psql -c option no longer implies --no-psqlrc with postgresql 9.6, it must be set explicitly.

Fixes #36787.

s0undt3ch commented:
How far back does `psql` support `--no-psqlrc`?

maximeguillet commented:
As far I found, option is already present in postgresql 7.1 (Release date: 2001-08-15)

https://www.postgresql.org/docs/7.1/static/app-psql.html

s0undt3ch commented:
I guess no one is running that far back these days ;)

maximeguillet committed:
Fix postgresql tests by adding --no-psqlrc option introduced by #36787.

cachedout commented:
@maximeguillet Postgres unit tests didn't like this change. Could you take a look?

maximeguillet committed:
Fix postgresql tests using position in the argument list of psql.

Some tests use position number in the argument list of psql command
to check their results. As #36787 add one element, next position
in the list must be used.

maximeguillet committed:
Fix one remaining postgresql tests linked to #36787.

maximeguillet commented:
All unit tests are OK now, I have adjusted the number of elements in the call arguments accordingly with the additional option --no-psqlrc
